wandb/sdk/lib/sock_client.py

```diff
@@ -64,6 +64,27 @@ def send_server_response(self, msg: Any) -> None:
             # things like network status poll loop, there might be a better way to quiesce
             pass
 
+    def send_and_recv(
+        self,
+        *,
+        inform_init: spb.ServerInformInitRequest = None,
+        inform_start: spb.ServerInformStartRequest = None,
+        inform_attach: spb.ServerInformAttachRequest = None,
+        inform_finish: spb.ServerInformFinishRequest = None,
+        inform_teardown: spb.ServerInformTeardownRequest = None
+    ) -> spb.ServerResponse:
+        self.send(
+            inform_init=inform_init,
+            inform_start=inform_start,
+            inform_attach=inform_attach,
+            inform_finish=inform_finish,
+            inform_teardown=inform_teardown,
+        )
+        response = self.read_server_response(timeout=1)
```

Comment:
Isn't 1 sec too much?

Reply:
Honestly, I don't have a good sense... I just used the same timeout as here: https://github.com/wandb/client/blob/master/wandb/sdk/interface/router_sock.py#L26

Reply:
@raubitsj left  a comment as discussed